### PR TITLE
Update the link to client-gen doc

### PR DESCRIPTION
**Pull request by caesarxuchao**
The doc itself also needs update, i'll create an issue.

### CODE REVIEW

File: staging/src/k8s.io/kube-gen/cmd/client-gen/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-See [generating-clientset.md](https://kubernetes.io/docs/devel/generating-clientset.md)
+See [generating-clientset.md](https://github.com/kubernetes/community/blob/master/contributors/devel/generating-clientset.md)
```
Comment (cblecker) [inline staging/src/k8s.io/kube-gen/cmd/client-gen/README.md]: https://git.k8s.io/community/contributors/devel/generating-clientset.md

Comment (caesarxuchao) [inline staging/src/k8s.io/kube-gen/cmd/client-gen/README.md]: Curious what's the benefit other than being shorter? Is it a more stable link?

Comment (cblecker) [inline staging/src/k8s.io/kube-gen/cmd/client-gen/README.md]: It should be stable if we ever move or have to redirect things. It's something @thockin created (https://github.com/kubernetes/k8s.io/pull/59) that handles the formatting redirection. Right now it's a noop, but it's preferable if we're updating it anyways.

Comment (caesarxuchao) [inline staging/src/k8s.io/kube-gen/cmd/client-gen/README.md]: Thanks. Updated.

**Pull request merged**